File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer2(py y java)/Bus.java
public class Bus {
    private int capacidad;
    private int pasajeros;
    private float costo;


    public Bus(int capacidad, float costo) {
        this.capacidad=capacidad;
        this.pasajeros=0;
        this.costo=costo;
    }

    // a) 
    public void subirPasajeros(int cantidad) {
        if (this.pasajeros+cantidad <= this.capacidad) {
            this.pasajeros+=cantidad;
        } else {
            System.out.println("No hay suficientes asientos");
        }
    }

    // b) 
    public float cobrarPasaje() {
        return this.pasajeros*this.costo;
    }

    // c) 
    public int asientosDisponibles() {
        return this.capacidad-this.pasajeros;
    }

    // d) 
    public static void main(String[] args) {
        Bus miBus=new Bus(40,1.5f);
        miBus.subirPasajeros(10); 
        System.out.println("fue cobradooo: "+miBus.cobrarPasaje()+" Bs"); 
        System.out.println("Asientos disponibles: "+miBus.asientosDisponibles()); 
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/POO/Ejer2(py y java)/Bus.py
class Bus:
    def __init__(self, capacidad, costo):
        self.capacidad=capacidad
        self.pasajeros=0
        self.costo=float(costo)

    # a)
    def subir_pasajeros(self, cantidad):
        if self.pasajeros+cantidad <= self.capacidad:
            self.pasajeros+=cantidad
        else:
            print("No hay suficientes asientos")

    # b)
    def cobrar_pasaje(self):
        return self.pasajeros*self.costo

    # c)
    def asientos_disponibles(self):
        return self.capacidad-self.pasajeros


# d)
miBus=Bus(40,1.5)
miBus.subir_pasajeros(10)  
print("Total cobrado:", miBus.cobrar_pasaje(),"Bs")  
print("Asientos disponibles:", miBus.asientos_disponibles())  
